standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5586,12 +5807,39 @@ public PartitionsResponse get_partitions_req(PartitionsRequest req)
   public List<Partition> get_partitions_with_auth(final String dbName,
       final String tblName, final short maxParts, final String userName,
       final List<String> groupNames) throws TException {
-    return get_partitions_ps_with_auth(dbName, tblName,
```

Comment:
We don't need to restore `get_partitions_with_auth_optional_schema`, the `GetPartitionsArgs` has contains all the context for fetching the partition